dnf5-plugins/config-manager_plugin/addrepo.cpp

```diff
@@ -0,0 +1,462 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "addrepo.hpp"
+
+#include "shared.hpp"
+
+#include <curl/curl.h>
+#include <fmt/format.h>
+#include <libdnf5/conf/const.hpp>
+#include <libdnf5/repo/file_downloader.hpp>
+#include <libdnf5/repo/repo_query.hpp>
+#include <libdnf5/utils/bgettext/bgettext-mark-domain.h>
+
+#include <filesystem>
+#include <regex>
+#include <string_view>
+
+namespace dnf5 {
+
+using namespace libdnf5::cli;
+
+namespace {
+
+const std::filesystem::path CFG_MANAGER_REPOS_OVERRIDE_FILENAME = "99-config_manager.repo";
+
+const std::filesystem::path CFG_MANAGER_REPOS_OVERRIDE_FILEPATH =
+    libdnf5::REPOS_OVERRIDE_DIR / CFG_MANAGER_REPOS_OVERRIDE_FILENAME;
+
+
+// Extracts a specific part of the URL from a URL string.
+std::string get_url_part(const std::string & url, CURLUPart what_part) {
```

Comment:
May I ask you to describe that it returns an empty string if part does not exist?

Reply:
Described.